master: Update pkg/testutils/release/cockroach_releases.yaml

cockroach-teamcity opened the pull request:
Update pkg/testutils/release/cockroach_releases.yaml with recent values.

Epic: None
Release note: None
Release justification: test-only updates

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146685)
<!-- Reviewable:end -->

rail commented:
bors r+

celiala commented:
re-running failed CI job: https://github.com/cockroachdb/cockroach/actions/runs/15101625494?pr=146685

celiala commented on pkg/sql/logictest/REPOSITORIES.bzl:
```diff
@@ -7,17 +7,11 @@ CONFIG_DARWIN_AMD64 = "darwin-10.9-amd64"
 CONFIG_DARWIN_ARM64 = "darwin-11.0-arm64"
 
 _CONFIGS = [
-    ("24.3.12", [
```
🤔 should the 24.3 configs be removed?

cc @rail

rail replied:
We will remove those when 24.3 is EOL.

cockroach-teamcity committed:
master: Update pkg/testutils/release/cockroach_releases.yaml

Update pkg/testutils/release/cockroach_releases.yaml with recent values.

Epic: None
Release note: None
Release justification: test-only updates
